### PR TITLE
Allow emitting verbose assembly

### PR DESCRIPTION
**Pull request by whitequark**
See the docstring for explanation.

### CODE REVIEW

File: llvmlite/binding/targets.py [hunk 1]
```diff
@@ -246,6 +246,13 @@ def add_analysis_passes(self, pm):
         """
         ffi.lib.LLVMPY_AddAnalysisPasses(self, pm)
 
+    def set_verbose(self, verbose):
```
Comment (pitrou) [inline llvmlite/binding/targets.py]: How about a more descriptive name? `set_asm_verbosity` perhaps?

File: llvmlite/tests/test_binding.py [hunk 2]
```diff
@@ -632,6 +632,7 @@ def test_emit_assembly(self):
         target_machine = self.target_machine()
         mod = self.module()
         ee = self.jit(mod, target_machine)
+        target_machine.set_verbose(True)
```
Comment (pitrou) [inline llvmlite/tests/test_binding.py]: I think we should test both the non-verbose and the verbose case (perhaps with different checks?).

Comment (whitequark) [inline llvmlite/tests/test_binding.py]: The exact structure of the assembly comments, including the character that starts the comment, is dependent on the selected target, so I'm not sure how. Besides there isn't much to break, except the function not being called correctly...

File: llvmlite/binding/targets.py (hunk 1 above)
Comment (whitequark) [inline llvmlite/binding/targets.py]: Ah, yes, will do.

**Comment by whitequark** [comment]
Updated

**Comment by pitrou** [comment]
Thank you! Can you add documentation for the new method in `docs/source/binding/targets.rst`?

**Comment by whitequark** [comment]
Done

File: docs/source/binding/targets.rst [hunk 3]
```diff
@@ -154,6 +154,11 @@ Classes
       a code object, suitable for use with the platform's linker.
       A bytestring is returned.
 
+   .. method:: set_asm_verbosity(module, is_verbose)
```
Comment (pitrou) [inline docs/source/binding/targets.rst]: Just a nit: the method doesn't take a *module* argument.

Comment (whitequark) [inline docs/source/binding/targets.rst]: Ah right, fixed.

**Comment by pitrou** [comment]
Thanks! AppVeyor is currently broken because of a `conda` issue. I'm merging this PR.

**Pull request merged**